components/renku_data_services/users/db.py

```diff
@@ -176,36 +177,45 @@ async def _get_user(self, id: str) -> UserInfo | None:
     @dispatch_message(events.UpdateOrInsertUser)
     async def update_or_insert_user(
         self, user_id: str, payload: dict[str, Any], *, session: AsyncSession | None = None
-    ) -> UserWithNamespaceUpdate:
+    ) -> UserInfoUpdate:
         """Update a user or insert it if it does not exist."""
         if not session:
             raise errors.ProgrammingError(message="A database session is required")
-        res = await session.execute(select(UserORM).where(UserORM.keycloak_id == user_id))
+        res = await session.execute(
+            select(UserORM).where(UserORM.keycloak_id == user_id).options(selectinload(UserORM.namespace))
+        )
         existing_user = res.scalar_one_or_none()
         if existing_user:
             return await self._update_user(session=session, user_id=user_id, existing_user=existing_user, **payload)
         else:
             return await self._insert_user(session=session, user_id=user_id, **payload)
 
-    async def _insert_user(self, session: AsyncSession, user_id: str, **kwargs: Any) -> UserWithNamespaceUpdate:
+    async def _insert_user(self, session: AsyncSession, user_id: str, **kwargs: Any) -> UserInfoUpdate:
         """Insert a user."""
         kwargs.pop("keycloak_id", None)
         kwargs.pop("id", None)
-        new_user = UserORM(keycloak_id=user_id, **kwargs)
+        slug = base_models.Slug.from_user(
+            kwargs.get("email"), kwargs.get("first_name"), kwargs.get("last_name"), user_id
+        ).value.lower()
```

Comment:
suggestion: This is not necessary since the `__init__` method of `Slug` already calls `lower` on the value when it sets it.